internal/injector/aspect/advice/code/template.go

```diff
@@ -117,18 +144,22 @@ func (t *Template) compile(ctx context.Context, chain *node.Chain) ([]dst.Stmt,
 	}
 	file, err := dec.Parse(buf.Bytes())
 	if err != nil {
-		return nil, fmt.Errorf("while parsing generated code: %w\n%q", err, numberLines(buf.String()))
+		return nil, fmt.Errorf("while parsing generated code: %w\n%s", err, numberLines(buf.String()))
 	}
 
-	body := file.Decls[0].(*dst.FuncDecl).Body
-	dot.placeholders.replaceAllIn(body)
+	decls := make([]dst.Decl, 0, len(file.Decls))
+	for _, decl := range file.Decls {
+		if decl, ok := decl.(*dst.GenDecl); ok && decl.Tok == token.IMPORT {
+			return nil, errors.New("code templates must not contain import declarations, use the imports map instead")
+		}
+		decls = append(decls, dot.placeholders.replaceAllIn(decl).(dst.Decl))
+	}
 
-	list := body.List
-	stmts := make([]dst.Stmt, len(list))
-	for i, node := range list {
-		stmts[i] = t.processImports(ctx, ctxFile, node).(dst.Stmt)
+	for i := range decls {
+		decls[i] = t.processImports(ctx, ctxFile, decls[i]).(dst.Decl)
```

Comment:
This looks like this is the only usage of `processImports`. Maybe making it return `dst.Decl` would make more sense than casting the result each time